### PR TITLE
Introduce flake attempts for TM testruns

### PR DESCRIPTION
**Pull request by OlegLoewen**
**What this PR does / why we need it**:
Introduce flake attempts for TM testruns

**Release note**:

```
Introduce flake attempts for TM testruns
```

### CODE REVIEW

File: pkg/apis/testmachinery/v1beta1/types.go [hunk 1]
```diff
@@ -135,6 +135,9 @@ type TestrunStatus struct {
 	// This value is calculated by (CompletionTime - StartTime)
 	Duration int64 `json:"duration,omitempty"`
 
+	// Tells whether testrun failed first and succeeded in a retry
+	Flaked bool `json:"flaked,omitempty"`
```
Comment (schrodit) [inline pkg/apis/testmachinery/v1beta1/types.go]: do not make it part of the testrun resource as this parameter is only known by the testrunner

maybe rather add it to the Run or directly to the metadata

File: pkg/testrunner/types.go [hunk 2]
```diff
@@ -117,6 +120,7 @@ type TestrunSummary struct {
 	Phase     argov1.NodePhase `json:"phase,omitempty"`
 	StartTime *metav1.Time     `json:"startTime,omitempty"`
 	Duration  int64            `json:"duration,omitempty"`
+	Flaked    bool             `json:"flaked,omitempty"`
```
Comment (schrodit) [inline pkg/testrunner/types.go]: remove see comment above

File: pkg/testrunner/testrunner.go [hunk 3]
```diff
@@ -54,19 +54,35 @@ func (rl RunList) Run(log logr.Logger, tmClient kubernetes.Interface, namespace,
...
+				}
+				rl[i].Error = err
+
+				if err == nil && tr.Status.Phase == tmv1beta1.PhaseStatusSuccess {
+					if flakeAttempt != 0 {
+						tr.Status.Flaked = true
+					}
+					// testrun was successful, break retry loop
+					break
+				} else {
```
Comment (schrodit) [inline pkg/testrunner/testrunner.go]: else in not necessary as we break out in the if branch anyways

Comment (OlegLoewen) [reply pkg/testrunner/testrunner.go]: if flakeAttempts > 1, the else branch ist required

File: pkg/apis/testmachinery/v1beta1/types.go (hunk 1 above)
Comment (OlegLoewen) [reply pkg/apis/testmachinery/v1beta1/types.go]: done

**Pull request merged**